### PR TITLE
Rename `*AttributesGetter` to `*AttributeGetter`

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**

### CODE REVIEW

**Review by trask — APPROVED** [review]
I support this change https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/9977#pullrequestreview-1758827171

> What do you think of renaming *AttributesGetter to *AttributeGetter? (since these don't operate on Attributes or AttributesBuilder)

but would like a second opinion from @breedx-splk @jack-berg @open-telemetry/java-instrumentation-approvers

**Comment by breedx-splk** [comment]
I'm a little torn, and it's hugely pedantic. 

These classes don't operate on `Attributes` instances nor do they return `Attributes` instances, so that's fine. The challenge is that they are used to get one or more attributes from another type...but in most cases there are several methods per interface, which means you use them to get several different _attributes_. Pedantic, I know, but I think plural might actually be more appropriate.

Maybe now isn't a great time to be suggesting this change, but here goes....

I kinda dislike that they're named after the way they're used (for attributes). There's nothing really attribute specific about the interfaces nor the implementations.  I view these all as type specific adapters, like the design pattern, and I read in a book one time that it's good to make your classes after patterns.

So this makes me wonder if we couldn't drop `Attribute/s` from the names entirely. So then things might rename like ....

* `ServerAttributesGetter` -> `ServerAddressPortRequestAdapter`
* `CodeAttributesGetter` -> `CodeExecutionRequestAdapter`
* `SqlClientAttributesGetter` -> `SqlStatementRequestAdapter`
* `MessagingAttributesGetter` -> `MessagingInfoAdapter` (or just keep the original name for this one)
* `DubboClientNetworkAttributesGetter` -> lol ok this one doesn't need to be a single class, it combines multiple responsibilities, bad example
* `PulsarNetClientAttributesGetter` -> `PulsarAddressPortRequestAdapter`
* etc

I really don't feel strongly about this tho, and like that `Getter` is shorter than `RequestAdapter` or `ResponseAdapter` or whatever.

**Comment by trask** [comment]
discussed in SIG meeting, keeping existing names

**Pull request closed without merging**
